cmd/oras/internal/option/remote.go

```diff
@@ -144,8 +145,12 @@ func (opts *Remote) NewRepository(reference string, common Common) (repo *remote
 	if err != nil {
 		return nil, err
 	}
-
-	repo.PlainHTTP = opts.PlainHTTP
+	switch host, _, _ := net.SplitHostPort(repo.Reference.Registry); host {
```

Comment:
`host` will be empty if `repo.Reference.Registry` is `localhost` since `net.SplitHostPort` will return error in this case.